cloud/pkg/cloudstream/streamserver.go

```diff
@@ -153,6 +154,14 @@ func (s *StreamServer) getMetrics(r *restful.Request, w *restful.Response) {
 	}()
 
 	sessionKey := strings.Split(r.Request.Host, ":")[0]
+	if forwardedURI := r.Request.Header.Get("X-Forwarded-Uri"); forwardedURI != "" {
+		if t := strings.Split(forwardedURI, "/"); strings.HasPrefix(forwardedURI, "/api/v1/nodes/") && len(t) > 6 {
+			sessionKey = t[4]
+			if ip, ok := s.tunnel.getNodeIP(sessionKey); ok {
+				r.Request.Host = fmt.Sprintf("%s:%d", ip, constants.ServerPort)
+			}
+		}
```

Comment:
For this request, the node ip doesn't exist in `r.Request.Host`?

Reply:
https://github.com/kubeedge/kubeedge/issues/2436#issuecomment-754580697 I add some explanation.😂

Reply:
Looks good, thanks for the fixing.